### ISSUE TITLE
couldn't reproduce the results

### ISSUE THREAD

**Issue by zhujunru854** [MiraldiLab/maxATAC#149]
Hello, it's still me. I have successfully run maxATAC train, predict, and benchmark for ARID3A, but I couldn't reproduce your results. For ARID3A, I used K562 cell data as the test set, and G12878, HepG2, and MCF-7 as the training set. Your results show maxATAC_AUPR as 0.41 and maxATAC_Precision_at_5%_Recall as 0.7715. However, my final results are maxATAC_AUPR of 0.153 and maxATAC_Precision_at_5%_Recall around 0.18. Strangely, I used the preprocessed data you provided, so the results should have been consistent. Could you help identify what might be wrong?

I downloaded the preprocessed data you provided from here. And I directly used these bw and bed files to generate my meta file:
![Image](https://github.com/user-attachments/assets/541aefe6-c308-44d5-a67b-a1f0a52734c2)
![Image](https://github.com/user-attachments/assets/daccdb15-9b27-495c-be7c-1cdd23475906)

This is the meta file
![Image](https://github.com/user-attachments/assets/eb5a8048-34fe-4961-90e3-c66c8ddae018)

Here are the commands and parameters used for running the process：
TRAIN
![Image](https://github.com/user-attachments/assets/9fcf79e5-7fed-4c9a-bb9e-05a6072c1423)
Here are the graphs of loss and Dice coefficient during training（It is evident that overfitting has occurred, and due to the use of batch normalization after each convolutional layer, the loss decreases significantly right from the beginning. Therefore, I plan to attempt reducing batch normalization and the learning rate, and introducing regularization to improve the results）
![Image](https://github.com/user-attachments/assets/61b15b65-a6ce-4d33-be4a-1a68a28bce12)
PREDICT
![Image](https://github.com/user-attachments/assets/1c3e7346-6da8-4348-b6e7-bedcd403e3da)
BENCHMARK
![Image](https://github.com/user-attachments/assets/3706ce3f-1d6f-483a-a11c-88ba7fbaed3b)
（Ignore rand_ratio=0.5; I actually used 0.3）

This is the output result file: ARID3A_rand0.3_binary_revcomp99_RR30_benchmark_chr1_200bp_PRC.tsv.
![Image](https://github.com/user-attachments/assets/9ed33671-0165-4a59-9847-8b96223caa1c)

Similarly, I also tested ATF2, and the results were similar.

**Comment by FaizRizvi** [comment]
We trained our models for 50 epochs with a rand_ratio of 0 for best results (this makes the model train on 100% peak-centric regions).[Please see the Training Routines part of the maxATAC paper and the accompanying supplemental figure S2]. Are you training on all the CTs (creating a full model) and testing in chr1?

**Comment by zhujunru854** [comment]
Hello, thank you for your response. Firstly, I have modified the rand_ratio to 0 and set the epoch to 50 as per your suggestion, but I don't think it has helped. Secondly, I haven't trained on all cell types—I only used the data from the cell types labeled as 'Train' in the meta file for training, and used the chr1 data from the cell types labeled as 'Test' for testing. Isn't this the correct approach?

**Comment by FaizRizvi** [comment]
It depends on what you want to accomplish. I have done the training in both scenarios (leaving a CT out from training a maxATAC model and training full maxATAC models). So make sure you are comparing your results to the correct results published in the paper. 

I have run training with the same parameters. Perhaps there is an error in your meta file?

**Comment by zhujunru854** [comment]
Hello, it's still me. I retrained ATF, and this time the results are very close to yours: when testing with the K562 cell, the AUPRC is 0.63, and the accuracy at 5% recall is 0.97. However, **the results for ARID3A are still poor,** which I suspect may be due to **low data quality.**
![Image](https://github.com/user-attachments/assets/463799f7-d4b0-49ab-a681-d56f33e34a52)

Secondly, I have another question: I want to train the model **on the GPU** to make the process faster. My TensorFlow, CUDA, and cuDNN are all installed correctly, and during training, there is a prompt indicating that my GPU is recognized. However, when checking with nvtop, the GPU utilization remains 0, and the training mainly runs on the CPU. Currently, each epoch takes over 10 minutes. I also tried setting multiprocess=True, but it causes an error.

So, I want to ask if you have run the model on a GPU and how the code should be modified?

**Comment by XiaotingChen** [comment]
It's great that you can re-train some of the models and get very close results. This indicates that this issue can be closed as it's no longer a valid description of the problem. 

I would like to kindly remind you that deep neural network models can have certain randomness in their training process, depending on how random states are set in the entire pipeline. You can refer to the following TensorFlow pages for certain adjustments to the existing codebase to have an improved control over this topic.

https://www.tensorflow.org/api_docs/python/tf/random/set_seed

To improve the GPU training and its utilization, please refer to the following pages.

 https://www.tensorflow.org/guide/gpu_performance_analysis
